neptune/management/internal/api.py

```diff
@@ -165,33 +181,51 @@ def create_project(
     workspace, name = extract_project_and_workspace(name=name, workspace=workspace)
     project_qualified_name = f"{workspace}/{name}"
 
-    try:
-        workspaces = (
-            backend_client.api.listOrganizations(**DEFAULT_REQUEST_KWARGS).response().result
-        )
-        workspace_name_to_id = {f"{f.name}": f.id for f in workspaces}
-    except HTTPNotFound:
-        raise WorkspaceNotFound(workspace=workspace)
+    workspace_name_to_id = _get_workspace_name_id_map(backend_client, workspace)
 
     if workspace not in workspace_name_to_id:
         raise WorkspaceNotFound(workspace=workspace)
 
+    workspace_id = workspace_name_to_id[workspace]
+
+    if key is None:
+        project_keys = _get_projects_keys_in_organization(workspace_id, api_token)
+        key = ProjectKeyGenerator(name, project_keys).get_default_project_key()
+
     params = {
         "projectToCreate": {
             "name": name,
             "description": description,
             "projectKey": key,
-            "organizationId": workspace_name_to_id[workspace],
+            "organizationId": workspace_id,
             "visibility": ProjectVisibilityDTO.from_str(visibility).value,
         },
         **DEFAULT_REQUEST_KWARGS,
     }
 
+    project_response = _create_project(backend_client, project_qualified_name, params)
+
+    return normalize_project_name(
+        name=project_response.result.name, workspace=project_response.result.organizationName
+    )
+
+
+@with_api_exceptions_handler
+def _get_workspace_name_id_map(backend_client, workspace):
     try:
-        response = backend_client.api.createProject(**params).response()
-        return normalize_project_name(
-            name=response.result.name, workspace=response.result.organizationName
+        workspaces = (
+            backend_client.api.listOrganizations(**DEFAULT_REQUEST_KWARGS).response().result
         )
+        workspace_name_to_id = {f"{f.name}": f.id for f in workspaces}
+    except HTTPNotFound:
+        raise WorkspaceNotFound(workspace=workspace)
```

Comment:
Looks like this `workspace` argument was not even used. And I don't see any possibilities of 404 being thrown in this endpoint.

Reply:
I just moved this code to separate method, I can do more cleanup if you want

Reply:
I know. But it's good to do some small cleanups when possible.

Reply:
Cleaned